Add GraphQL section

marktani opened the pull request:
Add a GraphQL section with an app using react-native-web-player.

I [included code blocks](https://github.com/dabbott/react-native-express/pull/25/files#diff-c7ae30dc1e4c8a81ff58b25decffd3b6R80) using ```, and they are not rendered as expected. How can I improve this?

![code-blocks-rne](https://cloud.githubusercontent.com/assets/1780597/23262319/978d0820-f9da-11e6-89b1-b82ff4495e68.png)

dabbott commented:
Ah yes, I forget code blocks are totally broken. I plan on adding support for syntax highlighted code blocks at some time, but *in the meantime*, can you try:

```js
${<pre><code>{
`query posts {
  allPosts {
    id
    imageUrl
    description
  }
}`
}</code></pre>}
```

It's ugly but it's not that big of a hack, and I can easily swap it out later for markdown syntax.

dabbott commented on src/pages/GraphQL.js:
```diff
@@ -0,0 +1,117 @@
+    description
+  }
+}
+\`\`\`
```
After reading this part, I would wonder "what is this syntax? how does it work?"

Can we add a little more detail like, `GraphQL schemas and queries are written in (...whatever it's called...). You can learn more about this language [here]. This language is powerful because (...why...? off the top of my head: easy to limit, filter, select, whatever)`

dabbott commented on src/pages/GraphQL.js:
```diff
@@ -0,0 +1,117 @@
+]
+
+const content = markdown(markdownOptions)`
+[GraphQL](http://graphql.org) is a query language that allows the efficient fetching or hierarchical data. Client applications simply state their data requirements which are then fulfilled by the GraphQL server. This dramatically speeds up app development time and decreases the coupling between front and backend.
```
`fetching or` => `fetching of`

dabbott commented:
Overall I like it a lot. I think it's way easier to follow this than when I tried to set up Apollo looking at the docs.

My main suggestion is, I think the order of the content assumes a bit of preexisting knowledge, and I'd like this to be tailored to people who have never heard of GraphQL.

I'd like it to read something like:

- Overview. (What is GraphQL? Mention you need a server + client library)
- What a GraphQL server is and does. (Mention the part about how we'll use Graphcool and why it's good)
- What a GraphQL client is and does. (Mention Apollo and describe it in a sentence or two. Mention that relay is another popular option, that's different from Apollo because X)
- Fetching data (How to make queries, link to learn more about the syntax)
- Example

E.g. the overview part you have now is good, but I'd also like to add something like `To use GraphQL, you need two things: a GraphQL-compatible server, and a GraphQL client.` which segues into a short section on GraphQL servers, then a short section on clients. You have a lot of this stuff already, just in a different order.

I don't think most people read what's below the example code, so I try to keep most things above 😛

marktani commented:
Thanks for the helpful feedback!

> My main suggestion is, I think the order of the content assumes a bit of preexisting knowledge, and I'd like this to be tailored to people who have never heard of GraphQL.

I agree! I adjusted the order, added a lot more information and also linked to some additional resources. What do you think? :slightly_smiling_face:

dabbott commented:
Looks great! 👍  😄 

I will merge and then fix the code block styles before we push it live.
